### PR TITLE
create endpoint to fetch top artists by year

### PR DESCRIPTION
**Pull request by opeyemi-paystack**

### CODE REVIEW

**Comment by ife-paystack** [comment]
@opeyemi-paystack, this means we'd have a model for every new year?

**Comment by opeyemi-paystack** [comment]
> @opeyemi-paystack, this means we'd have a model for every new year?

Yup. Makes sense because having a model per existing collection is already the norm.

**Comment by ife-paystack** [comment]
> > @opeyemi-paystack, this means we'd have a model for every new year?
> 
> Yup. Makes sense because having a model per existing collection is already the norm.

Makes sense, makes sense. My main concern is having to make code changes every year like [here](https://github.com/PaystackHQ/paystack-music-api/pull/48/files#diff-04d76ea14bad43311817aedbc71aa40261577f0c420829db8241eafe7a5528f0R10)

**Comment by opeyemi-paystack** [comment]
> > > @opeyemi-paystack, this means we'd have a model for every new year?
> > 
> > 
> > Yup. Makes sense because having a model per existing collection is already the norm.
> 
> Makes sense, makes sense. My main concern is having to make code changes every year like [here](https://github.com/PaystackHQ/paystack-music-api/pull/48/files#diff-04d76ea14bad43311817aedbc71aa40261577f0c420829db8241eafe7a5528f0R10)

There's no way to avoid it because a new collection has to be created each time (Unless you can figure out a way to tell what tracks belong to playlists in a particular year without that first aggregation). There'd just be a detailed process anyone can follow to create the PR for a new wrapped year, with no more than 5 items. Pretty basic.

File: controllers/index.js [hunk 1]
```diff
@@ -136,4 +136,22 @@ module.exports = {
   },
 
   webhook: (req, res) => { },
+
+  wrappedGetTopArtists: async (req, res) => {
+    try {
+      const limit = req.query.limit || 3;
+      const year = req.query.year || '2020';
```
Comment (ife-paystack) [inline controllers/index.js:143]: Let's remove 2020 and make year a required parameter from Joi.

**Comment by ife-paystack** [comment]
> > > > @opeyemi-paystack, this means we'd have a model for every new year?
> > > 
> > > 
> > > Yup. Makes sense because having a model per existing collection is already the norm.
> > 
> > 
> > Makes sense, makes sense. My main concern is having to make code changes every year like [here](https://github.com/PaystackHQ/paystack-music-api/pull/48/files#diff-04d76ea14bad43311817aedbc71aa40261577f0c420829db8241eafe7a5528f0R10)
> 
> There's no way to avoid it because a new collection has to be created each time (Unless you can figure out a way to tell what tracks belong to playlists in a particular year without that first aggregation). There'd just be a detailed process anyone can follow to create the PR for a new wrapped year, with no more than 5 items. Pretty basic.

I currently can't think of anything but the duplication of the tracks model and code for each new year bothers me a lot 😭 
At 2040, there'd be 20 models with the same schema and 20 lines of code defining years.

File: controllers/index.js (hunk 1 above)
Comment (opeyemi-paystack) [reply controllers/index.js:143]: Did it this way because I like the idea of having the first year as the default but sure.

**Comment by opeyemi-paystack** [comment]
> > > > > @opeyemi-paystack, this means we'd have a model for every new year?
> > > > 
> > > > 
> > > > Yup. Makes sense because having a model per existing collection is already the norm.
> > > 
> > > 
> > > Makes sense, makes sense. My main concern is having to make code changes every year like [here](https://github.com/PaystackHQ/paystack-music-api/pull/48/files#diff-04d76ea14bad43311817aedbc71aa40261577f0c420829db8241eafe7a5528f0R10)
> > 
> > 
> > There's no way to avoid it because a new collection has to be created each time (Unless you can figure out a way to tell what tracks belong to playlists in a particular year without that first aggregation). There'd just be a detailed process anyone can follow to create the PR for a new wrapped year, with no more than 5 items. Pretty basic.
> 
> I currently can't think of anything but the duplication of the tracks model and code for each new year bothers me a lot 😭
> At 2040, there'd be 20 models with the same schema and 20 lines of code defining years.

For that there should be a limit of 5 years. For every new year that goes beyond 5, the oldest one is removed.

File: controllers/index.js (hunk 1 above)
Comment (opeyemi-paystack) [reply controllers/index.js]: Done

**Review by ife-paystack — APPROVED** [review]

**Pull request merged**